File: PizzaDelivery/PizzaDelivery/Models/Order.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace PizzaDelivery.Models
{
    public class Order
    {
        public int Id { get; set; }
        public DateTime TimeIn { get; set; }
        public DateTime TimeOut { get; set; }
        public bool Delivered { get; set; }
        public Customer Customer { get; set; }
        public List<OrderItem> OrderItems { get; set; }

        public Order() { }
        public Order(Order copy)
        {
            Id = copy.Id;
            TimeIn = copy.TimeIn;
            TimeOut = copy.TimeOut;
            Delivered = copy.Delivered;
            Customer = new Customer(copy.Customer);
            //OrderItems = copy.OrderItems.Select(item => new OrderItem(item)).ToList();
        }
    }
}

File: PizzaDelivery/DeliveryApp/src/app/models/order.ts
import { Customer } from "./customer";
import { OrderItem } from "./orderitem";
import { Store } from "./store";

export class Order {
    id?: number;
    customerId?: number;
    storeId?: number;
    timeIn?: Date;
    timeOut?: Date;
    delivered?: boolean;
    customer?: Customer;
    store?: Store;
    orderItems?: OrderItem[];
}
